### PR TITLE
Prompt ChatGPT ports

### PR DESCRIPTION
**Pull request by loks0n**

### CODE REVIEW

File: dart/prompt_chatgpt/lib/utils.dart [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      missing.add(key);
+    }
+  }
+  if (missing.isNotEmpty) {
+    throw Exception('Missing required fields: ${missing.join(', ')}');
+  }
+}
+
+final String _dirname = Platform.script.toFilePath();
+final String staticFolder = '${Uri.file(_dirname).resolve('../static')}';
```
Comment (Meldiron) [inline dart/prompt_chatgpt/lib/utils.dart:20]: Lets move those to the top of the file

File: php/prompt-chatgpt/src/index.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<?php
+
+require(__DIR__ . '/../vendor/autoload.php');
+
+return function ($context) {
+    throw_if_missing($_ENV, ['OPENAI_API_KEY']);
+
+    if ($context->req->method === 'GET') {
+        return $context->res->send(get_static_file('index.html'), 200, [
+            'Content-Type' => 'text/html',
```
Comment (Meldiron) [inline php/prompt-chatgpt/src/index.php:10]: In some PRs you do this, but in some I think you also specify UTF8 charset. Let's make it consistent across all pls.

File: php/prompt-chatgpt/src/index.php [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        ]);
+    }
+
+    try {
+        throw_if_missing($context->req->body, ['prompt']);
+    } catch (Exception $e) {
+        return $context->res->json(['ok' => False, 'error' => $e->getMessage()], 400);
```
Comment (Meldiron) [inline php/prompt-chatgpt/src/index.php:17]: Not sure about uppercase False in PHP. lets do `false`

File: php/prompt-chatgpt/src/index.php [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+            'model' => 'gpt-3.5-turbo',
+            'max_tokens' => $_ENV['MAX_TOKENS'] ?: 150,
+            'messages' => [
+                ['role' => 'user', 'content' => $context->req->body['prompt']]
+            ],
+        ]);
+
+        $completion = $response['choices'][0]['message']['content'];
+        return $context->res->json(['ok' => True, 'completion' => $completion], 200, [
+            'Content-Type' => 'text/plain',
```
Comment (Meldiron) [inline php/prompt-chatgpt/src/index.php:33]: Is this nessessary header?

File: php/prompt-chatgpt/src/index.php [hunk 5]
```diff
@@ -0,0 +1,38 @@
+<?php
+
+require(__DIR__ . '/../vendor/autoload.php');
```
Comment (Meldiron) [inline php/prompt-chatgpt/src/index.php:3]: Dont we need to also require utils.php?

File: python/prompt_chatgpt/src/main.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
+import openai
+from utils import get_static_file, throw_if_missing
+import os
+
+
```
Comment (Meldiron) [inline python/prompt_chatgpt/src/main.py:5]: Double line break. also in utils.py

File: php/prompt-chatgpt/src/index.php [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+return function ($context) {
+    throw_if_missing($_ENV, ['OPENAI_API_KEY']);
+
+    if ($context->req->method === 'GET') {
+        return $context->res->send(get_static_file('index.html'), 200, [
```
Comment (Meldiron) [inline php/prompt-chatgpt/src/index.php:9]: Static folder missing in PHP

File: dart/prompt_chatgpt/lib/main.dart [hunk 8]
```diff
@@ -0,0 +1,40 @@
+import 'dart:io';
+import 'dart:async';
+import 'package:dart_openai/dart_openai.dart';
+import 'utils.dart';
+
+Future<dynamic> main(final context) async {
+  throwIfMissing(Platform.environment, ['OPENAI_API_KEY']);
+
+  if (context.req.method == 'GET') {
+    return context.res.send(getStaticFile('index.html'), 200,
```
Comment (Meldiron) [inline dart/prompt_chatgpt/lib/main.dart:10]: Static folder missing in Dart

File: python/prompt_chatgpt/src/main.py (hunk 6 above)
Comment (loks0n) [reply python/prompt_chatgpt/src/main.py:5]: This is the output of black formatter, I think we should keep it

**Review by Meldiron — APPROVED** [review]

**Pull request merged**